### PR TITLE
more volumes shorthand

### PR DESCRIPTION
**Pull request by ublubu**
#94 #77 

builds on #106 

@wlan0 

Still in progress: config-map, secret, downward-api, git, projected, persistent volumes

### CODE REVIEW

File: testdata/pods/pod_spec_with_volume_source_aws_ebs.short.yaml [hunk 1]
```diff
@@ -8,9 +8,9 @@ pod:
...
   volumes:
-  - fs_type: xfs
-    name: aws_ebs_test_volume
-    read_only: true
-    type: aws-elastic-block-store
-    volume_id: i4054053
+    aws_ebs_test_volume:
+      fs: xfs
+      ro: true
+      vol_name: i4054053
```
Comment (wlan0) [inline testdata/pods/pod_spec_with_volume_source_aws_ebs.short.yaml]: This should be called ID

File: testdata/pods/pod_spec_with_volume_source_azure_disk.short.yaml [hunk 2]
```diff
@@ -8,12 +8,11 @@ pod:
...
-    kind: Dedicated
-    name: azure_disk_test_volume
-    read_only: false
-    type: azure-disk
+    azure_disk_test_volume:
+      cache: rw
+      disk_uri: disk://uri.azure.disk
+      fs: xfs
+      kind: dedicated
+      vol_name: azure_disk_name
```
Comment (wlan0) [inline testdata/pods/pod_spec_with_volume_source_azure_disk.short.yaml]: disk_name: $disk_name

`vol_name` is confusing.

File: testdata/pods/pod_spec_with_volume_source_cinder.short.yaml [hunk 3]
```diff
@@ -8,9 +8,9 @@ pod:
...
   volumes:
-  - fs_type: xfs
-    name: test_volume
-    read_only: true
-    type: cinder
-    volume_id: cinderID
+    test_volume:
+      fs: xfs
+      ro: true
+      vol_name: cinderID
```
Comment (wlan0) [inline testdata/pods/pod_spec_with_volume_source_cinder.short.yaml]: vol_id instead of name everywhere?

Comment (ublubu) [reply testdata/pods/pod_spec_with_volume_source_cinder.short.yaml]: sgtm

**Comment by ublubu** [comment]
updated, @wlan0 
switched vol_name to vol_id
for azure_disk, moved vol_id to disk_name
for glusterfs, moved path to vol_id, moved vol_id to endpoints. (both fields are required) according to the docs, path is the volume id, so it makes more sense this way.

**Comment by wlan0** [comment]
Thanks, LGTM

**Pull request merged**